sdk/helper/scaleutils/node_drain_test.go

```diff
@@ -4,16 +4,38 @@
 package scaleutils
 
 import (
+	"context"
 	"errors"
 	"testing"
 	"time"
 
+	hclog "github.com/hashicorp/go-hclog"
 	multierror "github.com/hashicorp/go-multierror"
 	errHelper "github.com/hashicorp/nomad-autoscaler/sdk/helper/error"
 	"github.com/hashicorp/nomad/api"
+	"github.com/shoenig/test/must"
 	"github.com/stretchr/testify/assert"
 )
 
+type mockDrainer struct {
+	drain func(nodeID string, opts *api.DrainOptions,
+		q *api.WriteOptions) (*api.NodeDrainUpdateResponse, error)
+	monitor func(ctx context.Context, nodeID string,
+		index uint64, ignoreSys bool) <-chan *api.MonitorMessage
+	monitorCalled bool
+}
```

Comment:
It took me a while to figure out how to use this correctly and the newlines make it a little hard to read. Would it be possible to split out the funcs into their own const declaration, have a constructor func for `mockDrainer`, and some comments? I think this would help readability and future use.

Reply:
Done!